git-keeper-client/gkeepclient/client_configuration.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2016 Nathan Sommer and Ben Coleman
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import configparser
+import os
+
+
+# Store one module-level ClientConfiguration instance
+config_instance = None
+
+
+class ConfigurationError(Exception):
+    pass
+
+
+class ClientConfiguration:
+    """Parses client.cfg and stores the configuration values as attributes.
+
+    Public attributes:
+
+    Server-related attributes:
+        host - hostname of the server
+        username - the faculty member's username on the server
+        ssh_port - the port to use for SSH connections (defaults to 22)
+
+    """
+
+    def __init__(self):
+        """Parse the configuration file and initialize the attributes"""
+        self._config_path = \
+            os.path.expanduser('~/.config/git-keeper/client.cfg')
+
+        if not os.path.isfile(self._config_path):
+            error = '{0} does not exist'.format(self._config_path)
+            raise ConfigurationError(error)
+
+        self._parse_config_file()
+        self._initialize_server_attributes()
+
+    def _parse_config_file(self):
+        """Use a ConfigParser object to parse the configuration file"""
+
+        self._parser = configparser.ConfigParser()
+
+        try:
+            self._parser.read(self._config_path)
+        except configparser.MissingSectionHeaderError:
```

Comment:
Is this the only exception that could be raised here?  A brief scan of the docs shows there are a variety of other ones in the library.